include/NovelRT/Ecs/EcsUtils.h

```diff
@@ -20,11 +24,11 @@ namespace NovelRT::Ecs
      * @tparam TComponent The component type to get the ID for.
      * @return ComponentTypeId The numerical ID of the component type as an Atom.
      */
-    template<typename TComponent>[[nodiscard]] ComponentTypeId GetComponentTypeId() noexcept
+    template<typename TComponent> [[nodiscard]] ComponentTypeId GetComponentTypeId() noexcept
     {
-        static const Atom id = Atom::getNextComponentTypeId();
-        return id;
+        auto componentTypeId = GetComponentTypeIds()[typeid(TComponent)];
```

Comment:
MSVC and Clang differ in behavior here. For Clang, this was working as expected and `A.dll` and `B.exe` would share a storage slot for `T` if the method was marked `extern`.

For MSVC, however, it doesn't look for exports if `B.exe` directly contains a definition itself (which makes sense IMO, prefer local over external) but that causes `A.dll` and `B.exe` to have inconsistent world views for a given `T`.

This resolves that issue by using `RTTI`, specifically it uses an `unordered_map<type_index, ComponentTypeId>`. `typeid(T)` for a statically known, non-polymorphic `T` is a constant time lookup and so we don't expect any issues here around perf.

Reply:
Could we add the `std::is_trivially_copyable` trait check to this? Would make it consistent with `NovelRT::Ecs:SparseSet<T>` :D

Reply:
Done